scenarios/test_codegen/test/lib_tests/FetchState_test.res

```diff
@@ -1353,6 +1352,163 @@ describe("FetchState unit tests for specific cases", () => {
     )
   })
 
+  it("Should be fetching at head only when all partitions are fetching at head", () => {
+    let fetchState = FetchState.make(
+      ~staticContracts=[("ContractA", mockAddress1), ("ContractB", mockAddress2)],
+      ~dynamicContracts=[],
+      ~startBlock=0,
+      ~endBlock=None,
+      ~maxAddrInPartition=1,
+      ~isFetchingAtHead=false,
+    )
+
+    let q0 = FetchState.PartitionQuery({
+      partitionId: "0",
+      contractAddressMapping: ContractAddressingMap.make(),
+      fromBlock: 0,
+      toBlock: None,
+    })
+    let q1 = FetchState.PartitionQuery({
+      partitionId: "1",
+      contractAddressMapping: ContractAddressingMap.make(),
+      fromBlock: 0,
+      toBlock: None,
+    })
+
+    Assert.equal(fetchState.isFetchingAtHead, false)
+
+    let fetchStateWithResponse1 =
+      fetchState
+      ->FetchState.setQueryResponse(
+        ~query=q0,
+        ~newItems=[],
+        ~currentBlockHeight=10,
+        ~latestFetchedBlock=getBlockData(~blockNumber=10),
+      )
+      ->Result.getExn
+
+    Assert.equal(
+      fetchStateWithResponse1.isFetchingAtHead,
+      false,
+      ~message=`Only partition "0" caught up to head,
+      should wait for partition "1" to catch up to head as well`,
```

Comment:
Since partitions became independent, I noticed that `isFetchingAtHead` constantly changing between true and false, when I ran metri indexer with `ENVIO_MAX_PARTITION_CONCURRENCY=5`. These tests and fixes are to solve the problem.